src/Discord.Net.Core/Entities/Channels/ITextChannel.cs

```diff
@@ -1,11 +1,11 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Threading.Tasks;
 
 namespace Discord
 {
-    public interface ITextChannel : IMessageChannel, IMentionable, IGuildChannel
+    public interface ITextChannel : IMessageChannel, IMentionable, IGuildChannel, INestedChannel
```

Comment:
How should we deal with redundant interfaces? `INestedChannel` inherits `IGuildChannel` already.
1) Keep both
2) Remove `IGuildChannel` from here (and other redundant places)
3) Make `INestedChannel` not inherit `IGuildChannel` (makes hard accessing specific properties when casting, dont recommend)

Reply:
we could make `ITextChannel` inherit `INestedChannel` (which inherits `IGuildChannel`)

Reply:
I went with option #2 for now.